x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/TransportGetStackTracesAction.java

```diff
@@ -351,17 +366,34 @@ private void searchEventGroupedByStackTrace(
                 new TermsAggregationBuilder("group_by")
                     // 'size' specifies the max number of host ID we support per request.
                     .size(MAX_TRACE_EVENTS_RESULT_SIZE)
-                    .field("process.executable.name")
-                    // missing("") is used to include documents where the field is missing.
-                    .missing("")
+                    .field("Stacktrace.sampling_frequency")
+                    // missing(DEFAULT_SAMPLING_RATE) is used to include documents where the field is missing.
+                    .missing((long) DEFAULT_SAMPLING_FREQUENCY)
                     // 'execution_hint: map' skips the slow building of ordinals that we don't need.
                     // Especially with high cardinality fields, this makes aggregations really slow.
                     .executionHint("map")
-                    .subAggregation(groupByThreadName)
+                    .subAggregation(groupByExecutableName)
+                    .subAggregation(new SumAggregationBuilder("total_count").field("Stacktrace.count"))
             )
             .addAggregation(new SumAggregationBuilder("total_count").field("Stacktrace.count"))
             .execute(handleEventsGroupedByStackTrace(submitTask, client, responseBuilder, submitListener, searchResponse -> {
-                long totalCount = getAggValueAsLong(searchResponse, "total_count");
+                long maxSamplingFrequency = 0;
+
+                Terms samplingFrequencies = searchResponse.getAggregations().get("group_by");
+                for (Terms.Bucket samplingFrequencyBucket : samplingFrequencies.getBuckets()) {
+                    final double samplingFrequency = samplingFrequencyBucket.getKeyAsNumber().doubleValue();
+                    if (samplingFrequency > maxSamplingFrequency) {
+                        maxSamplingFrequency = (long) samplingFrequency;
+                    }
+                }
+
+                long totalCount = 0;
+                for (Terms.Bucket samplingFrequencyBucket : samplingFrequencies.getBuckets()) {
+                    InternalNumericMetricsAggregation.SingleValue count = samplingFrequencyBucket.getAggregations().get("total_count");
+                    final double samplingFrequency = samplingFrequencyBucket.getKeyAsNumber().doubleValue();
```

Comment:
Can we skip the scaling here if `samplingFrequency == maxSamplingFrequency` ?

Reply:
It would not make any difference except for superfluous special casing code.